mdtraj/geometry/distance.py

```diff
@@ -135,25 +135,40 @@ def compute_displacements(traj, atom_pairs, periodic=True, opt=True):
     return _displacement(xyz, pairs)
 
 
-def compute_center_of_mass(traj):
+def compute_center_of_mass(traj, select=None):
     """Compute the center of mass for each frame.
 
     Parameters
     ----------
     traj : Trajectory
         Trajectory to compute center of mass for
+    select : str, optional, default=all
+        a mdtraj.Topology selection string that
+        defines the set of atoms of which to calculate
+        the center of mass, the default is all atoms
 
     Returns
     -------
     com : np.ndarray, shape=(n_frames, 3)
          Coordinates of the center of mass for each frame
     """
+    com = np.empty((traj.n_frames, 3))
 
-    com = np.zeros((traj.n_frames, 3))
-    masses = np.array([a.element.mass for a in traj.top.atoms])
-    masses /= masses.sum()
+    if select is None:
+        masses = np.array([a.element.mass for a in traj.top.atoms])
+        masses /= masses.sum()
 
-    for i, x in enumerate(traj.xyz):
+        xyz = traj.xyz
+
+    else:
+        atoms_of_interest = traj.topology.select(select)
+
+        masses = np.array([a.element.mass for a in traj.top.atoms if a.index in atoms_of_interest])
+        masses /= masses.sum()
+
+        xyz = np.array(traj.xyz[:, atoms_of_interest], order='C')
```

Comment:
What is the reason for this extra copy? traj.xyz should already be c-ordered, right? (otherwise the copy should also be done on line `161`)
```suggestion
        xyz = traj.xyz[:, atoms_of_interest]
```

Reply:
I must admit that I copied this line from the code of the topology atom slicing method without thinking too much about it.
Actually the code might be a bit faster with a fortran ordering as later we are doing a matrix multiplicatio of the transpose?

one should see what is faster between having an array contiguous in memory but having to do this copy or having a not contiguous one but no copy needed

Reply:
Ok, I played around with timeit and your solution is the faster one

Reply:
The test failure looks legitimate, I think.

Reply:
Error looks like it came from the other changed line:
```python
>   masses = np.array([traj.top.atom[i].element.mass for i in atoms_of_interest])
E   TypeError: 'method' object is not subscriptable
```
should be fixed by [this suggestion](https://github.com/mdtraj/mdtraj/pull/1640#discussion_r614274707) (I always forget that `atom` is a function and not a `list` )